### ISSUE TITLE
App unexpectedly gets refreshed when a network change occurs

### ISSUE THREAD

**Issue by gauthierpetetin** [MetaMask/metamask-mobile#11659]
### What is this about?

[Slack thread](https://consensys.slack.com/archives/CBW7S9FSN/p1728286265701319) with more detail.

TLDR: This unexpected dapp refresh only occurs when switching to a network that was never used before.

### Scenario

- GIVEN As a user, I'm on a dapp (e.g. Phospos dapp) on the in-app browser
- WHEN I switch network
- THEN The dapp shall not refresh in the in-app browser

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

**Comment by Cal-L** [comment]
Could potentially be solved by removing the Drawer view

**Comment by alfeng6** [comment]
@Cal-L is this still an issue with all the perf improvements and new multichain work?

**Comment by Prithpal-Sooriya** [comment]
I'm closing as this is not reproducible - many changes have gone in the last 6 months that makes this hard to pinpoint strictly to assets.
